### ISSUE TITLE
[Question] 

### ISSUE THREAD

**Issue by Mert-Z** [renovatebot/config-help#286]
### Which Renovate are you using? CLI, App, or Pro
CLI (slef-hosted)

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps
Self hosted GitHub

### Have you checked the logs? Don't forget to include them if relevant
Yes, logs are not relevant.

### What is your question?
So we have the following dependencies in our pom.xml (just including only relevant parts);
```
...
<properties>
    ...
       <jackson.version>2.9.9</jackson.version>
    ...
</properties>
<dependencies>
    ...
       <dependency>
            <groupId>com.fasterxml.jackson.dataformat</groupId>
            <artifactId>jackson-dataformat-xml</artifactId>
            <version>${jackson.version}</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jdk8</artifactId>
            <version>${jackson.version}</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>${jackson.version}</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jsr310</artifactId>
            <version>${jackson.version}</version>
        </dependency>
    ...
</dependencies>
```

RenovateBot detected a new version for "jackson-databind" which is 2.9.9.1 and raised a PR as expected.

However changeset includes this;
```
<properties>
    ...
-        <jackson.version>2.9.9</jackson.version>
+        <jackson.version>2.9.9.1</jackson.version>
    ...
</properties>
```
eventually this causes an update of the other jackson dependencies which relies on "jackson.version" placeholder (as if they are grouped) and causes PR build to fail as those components don't have such a version released.

So can renovate just update the version of the related dependency? Is there some kind of limitation on renovate around sharing version properties? What would be the best practices if that's the case? Should we have individual version properties for the components?

Thanks

**Comment by Mert-Z** [comment]
I ran some additional tests:
1. created individual version placeholders and grouped dependencies in package rules -> No PR raised
2. kept shared version placeholder and grouped dependencies in package rules -> No PR raised

So if I configure groups for dependencies that share same version placeholder, renovatebot waits for all dependencies to release the same new version before raising PR.

So this brings in another question of whether renovatebot can group implicitly the dependencies that share same version placeholder?

**Comment by rarkins** [comment]
The fundamental problem to overcome is this:

> eventually this causes an update of the other jackson dependencies which relies on "jackson.version" placeholder (as if they are grouped) and causes PR build to fail as those components don't have such a version released.

The problem as I see it is that you're using a shared version (`jackson.version`) for dependencies that don't always release together and share a common version. This means you might miss out on a critical hotfix for one (e.g. `2.9.9.1`) because the others did not also release such a hotfix. I think you need to consider not doing this now that you have the possibility of using Renovate's automation to keep related dependencies in sync.

Can you tell me: if you were to replace `jackson.version` with the actual version (e.g. `2.9.9`) for each dependency, what would be the downside? My reasoning:

Upside: you can now update them independently without breaking the build, but still maintaining grouping (assuming the right packageRule is written to group)

Downside: it makes it harder to *manually* update versions if you are weren't relying on Renovate.

Any others?

> So if I configure groups for dependencies that share same version placeholder, renovatebot waits for all dependencies to release the same new version before raising PR.

That doesn't sound like our intentional behaviour. Are you guessing at that, or have some proof from the logs? You should be able to search logs based on the branch name.

> So this brings in another question of whether renovatebot can group implicitly the dependencies that share same version placeholder?

Right now we can group based on the `sourceUrl`, so if they are all released from the same repo/monorepo then they could be grouped. It would be a good future improvement to support grouping implicitly based on placeholder name.

I think though this talk of grouping doesn't solve the fundamental problem that you've used a common version for packages that don't always share the same latest version.

Cc @zharinov, we should discuss this as a feature request although I have a feeling it came up once or twice before

**Comment by Mert-Z** [comment]
> > So if I configure groups for dependencies that share same version placeholder, renovatebot waits for all dependencies to release the same new version before raising PR.
> 
> That doesn't sound like our intentional behaviour. Are you guessing at that, or have some proof from the logs? You should be able to search logs based on the branch name.

I executed few more tests to make sure and realised that renovate is creating the PR in both cases. So I guess renovatebot was set to ignore the new version while I was testing since I was testing in the same branch and there was a closed PR related to same version already. Anyways I tested in a clean fork of the same project and also tested again the same with setting renovabot to not ignore the new dependency and now it works just as expected.

I've also taken this as an opportunity to clean up pom. I still prefer using placeholders which can be easily accessible at the top of the pom, but I refactored to include individual placeholders for each component.

To sum up looks good now. But I also believe still worthwhile checking implicit grouping based on shared placeholder. Thanks.
